pkg/receive/writer.go

```diff
@@ -21,26 +22,38 @@ type Appendable interface {
 	Appender() (storage.Appender, error)
 }
 
+type TenantStorage interface {
+	TenantAppendable(string) (Appendable, error)
```

Comment:
nit: we could add the parameter name to help hint that the string is a tenant name